palette/moreland/convert_test.go

```diff
@@ -0,0 +1,189 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package moreland
+
+import (
+	"image/color"
+	"math"
+	"testing"
+)
+
+func TestLinearToS(t *testing.T) {
+	if linearToS(0.015299702) != 0.1298716701086684 {
+		t.Errorf("linearToS(0.015299702) should equal 0.1298716701086684 but equals %g", linearToS(0.015299702))
+	}
+}
+
+func TestLinearRGB(t *testing.T) {
+	xyz := cieXYZ{X: 0.128392403, Y: 0.128221351, Z: 0.408477452}
+	lrgb := xyz.rgb()
+	wantR, wantG, wantB := 0.015299702837399953, 0.1330700251971, 0.4127549680071
```

Comment:
With a helper (below), this could be converted into a table-driven test. Also the tests below.

```
func valueFor(value interface{}, field string) float64 {
	return reflect.ValueOf(value).FieldByName(field).Interface().(float64)
}
```